tests/test_linearop.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+from numpy.testing import assert_allclose
+np.random.seed(42)
+import pytest
+
+from mpi4py import MPI
+
+rank = MPI.COMM_WORLD.Get_rank()
+size = MPI.COMM_WORLD.Get_size()
+
+import pylops
+
+from pylops_mpi import MPILinearOperator, MPIBlockDiag, DistributedArray
+
+par1 = {'ny': 101, 'nx': 101, 'dtype': np.float64}
+par1j = {'ny': 101, 'nx': 101, 'dtype': np.complex128}
+par2 = {'ny': 301, 'nx': 101, 'dtype': np.float64}
+par2j = {'ny': 301, 'nx': 101, 'dtype': np.complex128}
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize("par", [(par1), (par1j), (par2), (par2j)])
+def test_linearop(par):
+    Op = pylops.MatrixMult(A=((rank + 1) * np.ones(shape=(par['ny'], par['nx']))).astype(par['dtype']))
+    BDiag_MPI = MPIBlockDiag(ops=[Op, ])
+    assert isinstance(BDiag_MPI, MPILinearOperator)
+    assert isinstance(BDiag_MPI.H, MPILinearOperator)
+    assert isinstance(BDiag_MPI.T, MPILinearOperator)
+    assert isinstance(BDiag_MPI * -3, MPILinearOperator)
+    assert isinstance(BDiag_MPI.conj(), MPILinearOperator)
+    assert isinstance(BDiag_MPI + BDiag_MPI, MPILinearOperator)
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize("par", [(par1), (par1j)])
+def test_square_linearop(par):
```

Comment:
Add short doctoring explaining what this test is

Reply:
Done